src/sth_database.js

```diff
@@ -334,17 +334,18 @@
           connection.db.createCollection(collectionName,
             function (err, collection) {
               if (!err && shouldStoreHash) {
-                storeCollectionHash(params, isAggregated, collectionName, function(err) {
-                  if (err) {
+                storeCollectionHash(params, isAggregated, collectionName, function(hashErr) {
+                  if (hashErr) {
                     // There was an error when storing the collection hash
                     // Do nothing
                     sthLogger.warn('Error when storing the hash generated as part of the collection name into the database', {
                       operationType: sthConfig.OPERATION_TYPE.DB_LOG
                     });
+                    return callback(hashErr, collection);
                   }
+                  return callback(err, collection);
                 });
-              }
-              if (err && err.message === 'collection already exists') {
+              } else if (err && err.message === 'collection already exists') {
```

Comment:
More than reasonable doubt :D , this is needed to protect the STH of a race condition which may arise mainly when many notifications are received. The asynchronous call to `connection.db.createCollection()` (using Mongoose) may inform that the collection does not exist, but once the callback is called and the collection is tried to be created it may have been created due to another notification (for the same collection) :)